Source/WebKit/UIProcess/Extensions/Cocoa/_WKWebExtensionStorageSQLiteStore.mm

```diff
@@ -351,6 +351,25 @@ - (DatabaseResult)_createFreshDatabaseSchema
     return result;
 }
 
+- (SchemaVersion)_migrateToCurrentSchemaVersionIfNeeded
+{
+    dispatch_assert_queue(_databaseQueue);
+
+    auto databaseSchemaVersion = self._databaseSchemaVersion;
+    if (databaseSchemaVersion == 2 && currentDatabaseSchemaVersion == 1) {
+        // Safari shipped with a database schema version of 2, but when migrating to WebKit, the version was
+        // incorrectly marked as 1. This mismatch would normally trigger a database schema reset, erasing
+        // all storage data. However, since the schema for version 2 (Safari) and version 1 (WebKit) are
+        // identical, we simply set the version and return the current version to avoid unnecessary data loss.
+
+        [self _setDatabaseSchemaVersion:currentDatabaseSchemaVersion];
```

Comment:
Do we want to set the database schema version to 2 in this case to avoid doing this work every time?

Also - what is `currentDatabaseSchemaVersion` in this method?

Reply:
I was confused because the previous file had `auto currentDatabaseSchemaVersion = self._currentDatabaseSchemaVersion;` - so I wasn't sure if it was missing here due to a refactor. That sounds good.